dev/com.ibm.ws.messaging.comms.client/src/com/ibm/ws/sib/netty/jfapchannel/NettyNetworkConnection.java

```diff
@@ -0,0 +1,330 @@
+
+	/** Trace */
+	private static final TraceComponent tc = SibTr.register(NettyNetworkConnection.class,
+			JFapChannelConstants.MSG_GROUP,
+			JFapChannelConstants.MSG_BUNDLE);
+
+	/** Log class info on load */
+	static
+	{
+		if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) SibTr.debug(tc, "@(#) SIB/ws/code/sib.jfapchannel.client.rich.impl/src/com/ibm/ws/sib/netty/jfapchannel/NettyNetworkConnection.java, SIB.comms, WASX.SIB, uu1215.01 1.2");
+	}
+
+	/** The virtual connection */
+
+	private Channel chan;
+
+	private String chainName;
+	
+	private boolean isInbound;
+	
+	private Map<String, Object> sslOptions;
+	
+	private NettyTlsProvider tlsProvider;
+
+	private NettyFramework nettyBundle;
+	
+	private BootstrapExtended bootstrap;
+
+
+
+	/**
+	 * @param bootstrap The Netty bootstrap object to create a channel from
+	 * @param chainName The chain name to which the channel belongs to
+	 * @throws FrameworkException 
+	 */
+	public NettyNetworkConnection(BootstrapExtended bootstrap, String chainName, NettyFramework nettyBundle, Map<String, Object> sslOptions, NettyTlsProvider tlsProvider, boolean isInbound) throws FrameworkException
+	{
+		if (TraceComponent.isAnyTracingEnabled() && tc.isEntryEnabled()) SibTr.entry(this, tc, "<init>" ,new Object[] {bootstrap, chainName});
+
+//		init(bootstrap);
+		this.chainName = chainName;
+		this.sslOptions = sslOptions;
+		this.isInbound = isInbound;
+		this.tlsProvider = tlsProvider;
+		this.nettyBundle = nettyBundle;
+		// TODO Check if we need to clone this
+		this.bootstrap = bootstrap;
+
+		if (TraceComponent.isAnyTracingEnabled() && tc.isEntryEnabled()) SibTr.exit(tc, "<init>", new Object[] {bootstrap, chainName});
+	}
+
+	public NettyNetworkConnection(Channel chan, String chainName, boolean isInbound)
+	{
+		if (TraceComponent.isAnyTracingEnabled() && tc.isEntryEnabled()) SibTr.entry(this, tc, "<init>", chan);
+
+		this.chan = chan;
+		this.isInbound = isInbound;
+		this.chainName = chainName;
+
+		if (TraceComponent.isAnyTracingEnabled() && tc.isEntryEnabled()) SibTr.exit(tc, "<init>", new Object[] {chan});
+	}
+
+	public void linkOutboundConnection(OutboundConnection conn) throws NettyException {
+		if (TraceComponent.isAnyTracingEnabled() && tc.isEntryEnabled()) SibTr.entry(this, tc, "linkOutboundConnection");
+		if(this.chan == null) {
+			throw new NettyException("Error linking connection appropriately");
+		}
+		if(chan.attr(NettyJMSClientHandler.CONNECTION_KEY).get() != null) {
+			if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) SibTr.debug(this, tc, "linkOutboundConnection","Connection was already set: "+chan.attr(NettyJMSClientHandler.CONNECTION_KEY).get());
+		}
+		chan.attr(NettyJMSClientHandler.CONNECTION_KEY).set(conn);
+		if (TraceComponent.isAnyTracingEnabled() && tc.isEntryEnabled()) SibTr.exit(this, tc, "linkOutboundConnection");
+	}
+
+
+	/**
+	 * @return Returns the channel connection.
+	 */
+	Channel getVirtualConnection()
```

Comment:
I think it is confusing to have the terms 'channel', 'channel connection' and 'virtual connection' mean the same thing - perhaps one day I can chat about why it makes sense to have all these synonyms here and then it will all seem obvious - but I suspect it might be better as one term?

Reply:
....but keep if mirroring existing code that works better for you..not trying to make extra work for no benefit.